src/Subscriptions/StorageManager.php

```diff
@@ -118,19 +118,42 @@ public function storeSubscriber(Subscriber $subscriber, string $topic): void
     /**
      * Delete subscriber.
      *
+     * @param  string  $channel
      * @return \Nuwave\Lighthouse\Subscriptions\Subscriber|null
      */
     public function deleteSubscriber(string $channel): ?Subscriber
     {
         $key = self::SUBSCRIBER_KEY.".{$channel}";
         $hasSubscriber = $this->cache->has($key);
-
         $subscriber = $this->cache->get($key);
 
         if ($hasSubscriber) {
+            $this->removeSubscriberFromTopic($subscriber, $key);
             $this->cache->forget($key);
         }
 
         return $subscriber;
     }
+    
+    /**
+     * Remove the subscriber from the topic they are subscribed to.
+     *
+     * @param  Subscriber $subscriber
+     * @param  string $subscriberKey
+     */
+    public function removeSubscriberFromTopic(Subscriber $subscriber, string $subscriberKey)
+    {
+        $key = strtoupper(Str::snake($subscriber->operationName));
```

Comment:
This won't work because a subscription can have a custom topic encoding/decoding scheme.

https://github.com/nuwave/lighthouse/blob/6552b798ab99790ca3344dad1b46046f9880b194/src/Schema/Types/GraphQLSubscription.php#L23-L45

Reply:
I thought this would be the case. Can dig into this more later.

Reply:
Storing the topic the subscriber before storing it now. Think that should do it?